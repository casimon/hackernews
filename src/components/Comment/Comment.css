.Comment {
  background-color: #f0f0f0;
  border-radius: 8px;
  padding: 20px;
  margin-bottom: 20px;
  font-size: 14px;
  font-style: italic;
  color: #222222;
}

.Comment div,
.Comment div p {
  font-size: 14px;
  font-style: italic;
  color: #222222;
  padding: 0;
  margin-bottom: 10px;
}

.Comment a {
  font-size: 14px;
  font-style: italic;
  color: #fb651e;
}
