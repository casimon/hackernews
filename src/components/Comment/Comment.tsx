import React from 'react'
import './Comment.css'

function createMarkup(text: any) {
  return { __html: text }
}

const Comment = ({ text }: any) => {
  return (
    <div className='Comment' dangerouslySetInnerHTML={createMarkup(text)} />
  )
}

export default Comment
