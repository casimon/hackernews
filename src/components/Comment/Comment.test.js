import React from 'react'

import { render } from '@testing-library/react'
import Comment from './Comment'

const init = (props) => render(<Comment {...props} />)

describe('Test comment component', () => {
  test('should render comment with no props', () => {
    const { container } = init()
    expect(container).toMatchSnapshot()
  })

  test('should render comment with props', () => {
    const { container } = init({
      text: 'Test',
      by: 'Simon Gamble',
    })
    expect(container).toMatchSnapshot()
  })
})
