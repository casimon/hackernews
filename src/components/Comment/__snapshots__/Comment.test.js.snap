// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test comment component should render comment with no props 1`] = `
<div>
  <div
    class="Comment"
  >
    <div />
    <p>
      By 
    </p>
  </div>
</div>
`;

exports[`Test comment component should render comment with props 1`] = `
<div>
  <div
    class="Comment"
  >
    <div>
      Test
    </div>
    <p>
      By 
      Simon Gamble
    </p>
  </div>
</div>
`;
